sktime/forecasting/compose/tests/test_pipeline.py

```diff
@@ -499,27 +499,148 @@ def test_forecastx_attrib_broadcast():
     model_1 = model.clone()
     model_1.fit(df[["c"]], X=df[["d", "e"]], fh=[1, 2, 3])
 
-    assert hasattr(model_1, "forecaster_X_")
-    assert isinstance(model_1.forecaster_X_, NaiveForecaster)
-    assert model_1.forecaster_X_.is_fitted
+    assert not hasattr(model_1, "forecaster_X_")
```

Comment:
could you make two versions of this, one for the "has forecaster_X_" and one "has not" case?

Reply:
Can you please tell what do you want me to add?

I thought I covered all cases in `test_forecastx_skip_forecaster_X_fitting_logic` and `test_forecastx_flow_known_unknown_columns` (across parametrisation)?

Reply:
ah, my mistake. I was confused since `test_forecastx_skip_forecaster_X_fitting_logic` introduces the coverage that you are removing from the first test. All is fine.